vault/resource_gcp_auth_backend.go

```diff
@@ -261,8 +268,8 @@ func gcpAuthBackendUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 
 	data := map[string]interface{}{}
 
-	if v, ok := d.GetOk(consts.FieldCredentials); ok {
-		data[consts.FieldCredentials] = v
+	if d.HasChange(consts.FieldCredentials) {
```

Comment:
Ah good catch! I'm guessing this is what was causing the credentials to be written every time, causing issues when we had already rotated them. Thanks for the fix!